pkg/network/ovn_kubernetes.go

```diff
@@ -327,6 +327,7 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	data.Data["DNS_NAME_RESOLVER_ENABLE"] = featureGates.Enabled(apifeatures.FeatureGateDNSNameResolver)
 	data.Data["OVN_NETWORK_SEGMENTATION_ENABLE"] = featureGates.Enabled(apifeatures.FeatureGateNetworkSegmentation)
 	data.Data["OVN_OBSERVABILITY_ENABLE"] = featureGates.Enabled(apifeatures.FeatureGateOVNObservability)
+	data.Data["OVN_ROUTE_ADVERTISEMENTS_ENABLE"] = c.RouteAdvertisements == operv1.RouteAdvertisementsEnabled
```

Comment:
Shall we also check the featureGate here?

Reply:
I don't see the need. The API is hidden if the feature gate is not set.

Reply:
hmm then how come we check other features by feature gate here and not just assume the API is not available?

Reply:
because they are using the feature gate to enable or disable the feature itself

for routeadvertisements we have a dedicated flag